src/artemis/devices/fast_grid_scan.py

```diff
@@ -176,13 +176,16 @@ def kickoff(self) -> StatusBase:
 
         def check_valid_and_scan():
             try:
-                self.log.info("Waiting on position counter reset and valid settings")
+                self.log.debug("Waiting on position counter reset and valid settings")
                 while self.is_invalid() or not self.position_counter.get() == 0:
                     time.sleep(0.1)
                 self.log.debug("Running scan")
-                running = SubscriptionStatus(self.status, lambda value: value == 1)
-                run_requested = self.run_cmd.set(1)
-                (run_requested and running).wait()
+                running = SubscriptionStatus(
+                    self.status, lambda old_value, value, **kwargs: value == 1
+                )
```

Comment:
If you wait until my eiger code is merged, there's a util included (await_value in status.py) that will create this object for you and let you put the wait on the same line - await_value(pv, value).wait(timeout)